src/commonControls/numberControl/NumberControlBuiltIns.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import i18next from 'i18next';
+import { ControlInput } from '../..';
+import { NumberControlState } from './NumberControl';
+
+export namespace NumberControlBuiltIns {
+    export function defaultValidationFailureText(): (value?: number) => string {
+        return (value?: number) => i18next.t('NUMBER_CONTROL_DEFAULT_APL_INVALID_VALUE', { value });
+    }
+
+    export const confirmMostLikelyMisunderstandingInputs = (state: NumberControlState, input: ControlInput) =>
+        defaultMostLikelyMisunderstandingFunc(state.value) !== undefined;
+
+    export function defaultMostLikelyMisunderstandingFunc(value: number): number | undefined {
+        switch (value) {
```

Comment:
Should you use a map instead of switch? Also, perhaps an enum instead of magic numbers?

Reply:
Yes, I can use a map instead of switch, but enums aren't really required for integers return value right? Since its a unique pre-defined pairs to resolve ambiguity

Reply:
Enum isn't required but I have no idea what the numbers are. Giving names to those numbers would help with understanding this code.

Reply:
I see, these are the subset of ambiguous pairs on integers where NLU can misinterpret or common inputs which could be misheard.
I have added a small comment on map definition and also examples is mentioned its on defined prop `mostLikelyMisunderstanding`.